proposals/dao/fip_78a.ts

```diff
@@ -0,0 +1,130 @@
+import hre, { ethers, artifacts } from 'hardhat';
+import { expect } from 'chai';
+import {
+  DeployUpgradeFunc,
+  NamedAddresses,
+  SetupUpgradeFunc,
+  TeardownUpgradeFunc,
+  ValidateUpgradeFunc
+} from '@custom-types/types';
+import { getImpersonatedSigner, time } from '@test/helpers';
+import { forceEth } from '@test/integration/setup/utils';
+
+const e18 = (x) => ethers.constants.WeiPerEther.mul(x);
+
+const fipNumber = '78a';
+const DELEGATE_ANGLE = '0x6ef71cA9cD708883E129559F5edBFb9d9D5C6148';
+
+// Do any deployments
+// This should exclusively include new contract deployments
+const deploy: DeployUpgradeFunc = async (deployAddress: string, addresses: NamedAddresses, logging: boolean) => {
+  const angleDelegatorFactory = await ethers.getContractFactory('AngleDelegatorPCVDeposit');
+  const angleDelegatorPCVDeposit = await angleDelegatorFactory.deploy(addresses.core, DELEGATE_ANGLE);
+  await angleDelegatorPCVDeposit.deployTransaction.wait();
+  logging && console.log('angleDelegatorPCVDeposit: ', angleDelegatorPCVDeposit.address);
+
+  // Create a new agEUR-FEI Uniswap PCVDeposit that does not stake on the old staking rewards contract
+  const uniswapPCVDepositFactory = await ethers.getContractFactory('UniswapPCVDeposit');
+  const agEurUniswapPCVDeposit = await uniswapPCVDepositFactory.deploy(
+    addresses.core,
+    addresses.angleAgEurFeiPool, // Uniswap-v2 agEUR/FEI pool
+    addresses.uniswapRouter, // UNiswap-v2 router
+    addresses.chainlinkEurUsdOracleWrapper,
+    ethers.constants.AddressZero,
+    '30' // max. 0.3% slippage
+  );
+  await agEurUniswapPCVDeposit.deployTransaction.wait();
+  logging && console.log('New agEUR/FEI Uniswap PCVDeposit:', agEurUniswapPCVDeposit.address);
+
+  const gaugeLensFactory = await ethers.getContractFactory('GaugeLens');
+  const gaugeLensAgEurUniswapGauge = await gaugeLensFactory.deploy(
+    addresses.angleGaugeUniswapV2FeiAgEur,
+    angleDelegatorPCVDeposit.address
+  );
+  await gaugeLensAgEurUniswapGauge.deployTransaction.wait();
+  logging && console.log('gaugeLensAgEurUniswapGauge:', gaugeLensAgEurUniswapGauge.address);
+
+  const uniswapLensFactory = await ethers.getContractFactory('UniswapLens');
+  const uniswapLensAgEurUniswapGauge = await uniswapLensFactory.deploy(
+    gaugeLensAgEurUniswapGauge.address, // deposit to read from
+    addresses.core, // core
+    addresses.chainlinkEurUsdOracleWrapper, // oracle
+    ethers.constants.AddressZero, // no backup oracle
+    false // fei is token1 in the pool
```

Comment:
With previous comment above, I'd try to push this `_isFeiToken0` bool down and have the contract programmatically figure it out